File: app/Models/SalesInvoice.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class SalesInvoice extends Model
{
    use HasFactory;

    protected $fillable = [
        'branch_id',
        'customer_id',
        'date',
        'total',
        'status',
        'payment_method',
        'note',
        'created_by',
    ];

    public function branch()
    {
        return $this->belongsTo(Branch::class);
    }

    public function items()
    {
        return $this->hasMany(SalesInvoiceItem::class);
    }

    public function creator()
    {
        return $this->belongsTo(User::class, 'created_by');
    }
    public function returns()
    {
        return $this->hasMany(SalesReturn::class);
    }

    public function customer()
    {
        return $this->belongsTo(Customer::class);
    }
}

File: desktop/lib/features/sales/domain/entities/sales_invoice.dart
import 'package:desktop/features/sales/domain/entities/sales_invoice_item.dart';

class SalesInvoice {
  final String id;
  final int branchId;
  final String? customerId;
  final String? customerName;
  final String date;
  final double total;
  final String status; // 'مدفوع', 'معلق', 'ملغى'
  final String paymentMethod; // 'نقدا', 'بطاقة', 'أخرى'
  final String? note;
  final int createdBy;
  final List<SalesInvoiceItem> items;
  final bool isSynced;
  final String? createdAt;

  const SalesInvoice({
    required this.id,
    required this.branchId,
    this.customerId,
    this.customerName,
    required this.date,
    required this.total,
    required this.status,
    required this.paymentMethod,
    this.note,
    required this.createdBy,
    required this.items,
    this.isSynced = false,
    this.createdAt,
  });
}
